R (#498)
Codecademy/docs: content/r/concepts/variables/variables.md

---
Title: 'Variables'
Description: 'A variable is used to store data that can be accessed later by subsequent code.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Variables'
  - 'Integers'
  - 'Booleans'
  - 'Strings'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

> A variable is used to store data that can be accessed later by subsequent code. In `R` there are no variable "declaration" commands.
> Variables are instead created with the assignment operator, `<-`. (The more familiar assignment operator, `=`, can be used instead, but is discouraged.)

Reviewer: ```suggestion
A variable is used to store data that can be accessed later by subsequent code. In R, there are no variable "declaration" commands. Instead, they are created with the assignment operator, `<-`. (The more familiar assignment operator, `=`, can be used instead, but is discouraged.)
```

I don't think it is necessary to wrap "R" in back-ticks this isn't the case in other files like `r.md`.

```r
# this creates a variable containing the string "value"
variable <- "value"
```

> The assignment operator can be chained together to initialize multiple variables at once:

Reviewer: By "chained" do we mean that all the `varx` variables are assigned the same value of 0? Or does it mean that `var3` is directly set to 0, `var2` is set to `var3`, and `var1` is set to `var2`?

Author: @Dusch4593: I'm not sure of the exact mechanics of this syntax, but it would make sense that the interpreter does the latter. Since the result is effectively the same, do we need to go into the distinction?

```r
var1 <- var2 <- var3 <- 0
```

> In `R`, putting a variable name by itself on a line has the effect of printing the value of that variable:

Reviewer: ```suggestion
In R, writing a defined variable name by itself on a line has the effect of printing its associated value:
```

```r
score <- 100

# This line:
score # Output 100

# Has the same effect as this line:
print(score) # Output 100
```

> ## Variable Naming Rules

Reviewer: ```suggestion
## Variable Names
```

> The name of an `R` variable must adhere to the following constraints:

Reviewer: ```suggestion
R variables must adhere to the following naming conventions:
```

- The name is a combination of letters, digits, period(.) and underscore(\_).
- It must start with a letter or a period.
- If it starts with a period, the second character cannot be a number.
- It cannot start with a number or an underscore.
- Variable names are case-sensitive.

> - Reserved words (TRUE,FALSE,print...) cannot be used for variable names.

Reviewer: ```suggestion
- Reserved words (TRUE, FALSE, print, etc.) cannot be used as variable names.
```

```r

> # Examples of valid variable names in R

Reviewer: Let's move this sentence above the code snippet and refactor to something like 

```
## Examples

In the snippet below are examples of valid variable names in R:

```

---
Title: 'Variables'
Description: 'A variable is used to store data that can be accessed later by subsequent code.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Variables'
  - 'Integers'
  - 'Booleans'
  - 'Strings'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

A variable is used to store data that can be accessed later by subsequent code. In R, there are no variable "declaration" commands. Instead, they are created with the assignment operator, `<-`. (The more familiar assignment operator, `=`, can be used instead, but is discouraged.)

```r
# this creates a variable containing the string "value"
variable <- "value"
```

The assignment operator can be chained together to initialize multiple variables at once:

```r
var1 <- var2 <- var3 <- 0
```

In R, writing a defined variable name by itself on a line has the effect of printing its associated value:

```r
score <- 100

# This line:
score # Output 100

# Has the same effect as this line:
print(score) # Output 100
```

## Variable Names

R variables must adhere to the following naming conventions:

> - The name is a combination of letters, digits, period(.) and underscore(\_).

Author: @caupolicandiaz: The underscore needs to be escaped or the validation complains. Followed the other suggestion.
